templates/server-deployment.yaml

```diff
@@ -31,14 +31,14 @@ spec:
         - name: {{ .Chart.Name }}
           securityContext:
             {{- toYaml .Values.securityContext | nindent 12 }}
-          image: "{{ .Values.server.spec.tamplate.spec.containers.image.name }}:{{ .Values.server.spec.tamplate.spec.containers.image.tag }}"
+          image: "{{ index .Values.server.spec.template.spec.containers.image.name }}:{{ .Values.server.spec.template.spec.containers.image.tag }}"
```

Comment:
NIT: the path here and in your other deployments is not the same level of nesting (for example for worker it's `.Values.worker.repository`) it's not super critical but it's easier to find things if the values have similar structure across the various deployments